src/gnatss/ops/io.py

```diff
@@ -184,8 +182,13 @@ def gather_files_all_procs(
             # if we are loading solution from cache
             continue
 
-        if proc == "main" or hasattr(config, proc):
+        if hasattr(config, proc):
             all_files_dict.update(gather_files(config, proc, mode))
+
+    if not all_files_dict:
+        msg = "Error: gather_files_all_procs() failed to parse input data."
+        raise ValueError(msg)
```

Comment:
```suggestion
    if not all_files_dict:
        avail_procs = ', '.join(constants.DEFAULT_CONFIG_PROCS)
        msg = "Could not find valid process in configuration. Valid options: {avail_procs}."
        raise ValueError(msg)
```

This is great, however I'd make the error message a little more clear and explicit. I'd suggest adding a test for this. To trigger it, simply remove the two processing attributes in the input config object, that should trigger this error. I know it's a little bit of work, but I think it's a good exercise to creating test and really thinking about what is happening within this section of the software 😄 Let me know if you have any questions!

Reply:
I'll see what I can put together in the next day or so and then reach back out.